config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/ListenPortEntityBuilder.java

```diff
@@ -101,20 +101,18 @@
     public List<Entity> build(Bundle bundle, BundleType bundleType, Document document) {
         switch (bundleType) {
             case DEPLOYMENT:
-                final Stream<Entity> userPorts = bundle.getListenPorts().entrySet().stream()
-                        .map(listenPortEntry -> buildListenPortEntity(bundle, listenPortEntry.getKey(), listenPortEntry.getValue(), document));
+                return bundle.getListenPorts().entrySet().stream()
+                        .map(listenPortEntry -> buildListenPortEntity(bundle, listenPortEntry.getKey(), listenPortEntry.getValue(), document)).collect(toList());
+            case ENVIRONMENT:
+                final Stream<Entity> userPorts = bundle.getListenPorts().entrySet().stream().map(listenPortEntry ->
+                        buildListenPortEntity(bundle, listenPortEntry.getKey(), listenPortEntry.getValue(), document));
                 final Stream<Entity> defaultPorts = DEFAULT_PORTS.entrySet().stream().filter(p -> bundle.getListenPorts().values().stream().noneMatch(up -> up.getPort() == p.getValue().getPort()))
```

Comment:
Just want to double check if the filter in default port is necessary or can we just map the default ports directly?

Reply:
I think this filtering exists to avoid remapping the default port if it was already specified by the user, for any reason.

Reply:
Ya, it's there incase there already are listen ports specified with same default port numbers in the environment bundle. If there are the defaults aren't added to the environment bundle. We could potentially also check the dependency bundle using the code below. but that might be overkill:
```java
                final Stream<Entity> defaultPorts = DEFAULT_PORTS.entrySet().stream()
                        .filter(p ->
                                bundle.getListenPorts().values().stream().noneMatch(up -> up.getPort() == p.getValue().getPort())
                                        && bundle.getDependencies().stream().noneMatch(b -> b.getListenPorts().values().stream().noneMatch(up -> up.getPort() == p.getValue().getPort())))
                        .map(listenPortEntry -> {
                            Entity entity = buildListenPortEntity(bundle, listenPortEntry.getKey(), listenPortEntry.getValue(), document);
                            entity.setMappingAction(NEW_OR_EXISTING);
                            return entity;
                        });
```